lib/factory_bot/strategy_syntax_method_registrar.rb

```diff
@@ -29,7 +29,10 @@ def define_list_strategy_method
           raise ArgumentError, "count missing for #{strategy_name}_list"
         end
 
-        Array.new(amount) { send(strategy_name, name, *traits_and_overrides, &block) }
+        Array.new(amount) do |i|
+          curried_block = block&.arity == 2 ? ->(record) { block.call(record, i) } : block
```

Comment:
code climate is complaining about the complexity of `define_list_strategy_method`, and I usually take that seriously.

What do you think about changing this to:

```suggestion
          block_with_index = StrategySyntaxMethodRegistrar.with_index(block, i)
```

Where `StrategySyntaxMethodRegistrar.with_index(block, i)` would be defined at the top of the class as:

```rb
    def self.with_index(block, index)
      if block&.arity == 2
        ->(instance) { block.call(instance, index) }
      else
        block
      end
    end
```

Note that I expanded the ternary to match the [thoughtbot style guide](https://github.com/thoughtbot/guides/tree/master/style/ruby). I also renamed `record` to `instance`, which is more in line with our naming elsewhere.

Reply:
Sounds great, works great. Done!